internal/controllers/configuration/kongupstreampolicy_utils.go

```diff
@@ -130,23 +133,13 @@ func (r *KongUpstreamPolicyReconciler) buildServicesStatus(ctx context.Context,
 			return nil, err
 		}
 
-		for _, httpRoute := range httpRoutes.Items {
-			for _, rule := range httpRoute.Spec.Rules {
-				if len(rule.BackendRefs) == 0 {
-					continue
-				}
-				for _, br := range rule.BackendRefs {
-					serviceRef := backendRefToServiceRef(httpRoute.Namespace, br.BackendRef)
-					if serviceRef == "" {
-						continue
-					}
-					if _, ok := mappedServices[serviceRef]; !ok {
-						serviceStatus.data.acceptedCondition.Status = metav1.ConditionFalse
-						serviceStatus.data.acceptedCondition.Reason = string(gatewayapi.PolicyReasonConflicted)
-						mappedServices[serviceKey] = serviceStatus
-					}
-				}
-			}
+		_, hasConflict := lo.Find(httpRoutes.Items, func(httpRoute gatewayapi.HTTPRoute) bool {
```

Comment:
I think we can use `return lo.ContainsBy(backendRefsUsedWithThisService,...)` if we only want to get the `true/false` result but not the certain item to find.